app/commands/cart_services/prep_cart_for_payment.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+#
+# Copyright 2021 Victoria Garcia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class CartServices::PrepCartForPayment
+
+  MEMBERSHIP = CartItem::MEMBERSHIP
+
+  attr_reader :our_cart, :amount_to_charge, :good_to_go, :processing_cart
+
+  def initialize(cart_obj)
+    @original_cart_object = cart_obj
+    @transaction_bin = identify_transaction_bin(cart_obj)
+    @user = note_user(cart_obj)
+    @amount_to_charge = 0
+    @good_to_go = false
+    @initial_item_count = initial_purchasable_item_count(cart_obj)
+    @holdable_types_created = {reservation: false}
+  end
+
+  def call
+    prep_cart_items
+    if confirm_all_items_prepped
+      @good_to_go = true
+    else
+      @amount_to_charge = 0
+    end
+    {good_to_go: @good_to_go, amount_to_charge: @amount_to_charge, holdable_types_made: @holdable_types_created }
+  end
+
+  private
+
+  def identify_transaction_bin(cart_o)
```

Comment:
What does `cart_o` mean?

Reply:
Cart Object.  That nomenclature is left over from when I was refactoring and wanted this method to take either a Cart or a CartChassis.  Now that such is no longer true, I can change that nomenclature.